src/providers/mssql/qgsmssqlconnection.cpp

```diff
@@ -167,6 +167,30 @@ void QgsMssqlConnection::setGeometryColumnsOnly( const QString &name, bool enabl
   settings.setValue( "/MSSQL/connections/" + name + "/geometryColumnsOnly", enabled );
 }
 
+bool QgsMssqlConnection::extentInGeometryColumns( const QString &name )
+{
+  QgsSettings settings;
+  return settings.value( "/MSSQL/connections/" + name + "/extentInGeometryColumns", false ).toBool();
+}
+
+void QgsMssqlConnection::setExtentInGeometryColumns( const QString &name, bool enabled )
+{
+  QgsSettings settings;
+  settings.setValue( "/MSSQL/connections/" + name + "/extentInGeometryColumns", enabled );
+}
+
+bool QgsMssqlConnection::primaryKeyInGeometryColumns( const QString &name )
+{
+  QgsSettings settings;
+  return settings.value( "/MSSQL/connections/" + name + "/primaryKeyInGeometryColumns", false ).toBool();
+}
+
+void QgsMssqlConnection::setPrimaryKeyInGeometryColumn( const QString &name, bool enabled )
```

Comment:
For consistency with other functions:
```suggestion
void QgsMssqlConnection::setPrimaryKeyInGeometryColumns( const QString &name, bool enabled )
```